consensus_pbft.go

```diff
@@ -602,35 +593,20 @@ func (p *Pbft) runRoundChangeState(ctx context.Context) {
 		}
 
 		// we only expect RoundChange messages right now
-		num := p.state.AddRoundMessage(msg)
-
-		//if voting power is disabled - count number of votes
-		if !p.IsVotingPowerEnabled() {
-			if num == p.state.NumValid() {
-				// start a new round inmediatly
-				p.state.SetCurrentRound(msg.View.Round)
-				p.setState(AcceptState)
-			} else if num == p.state.MaxFaultyNodes()+1 {
-				// weak certificate, try to catch up if our round number is smaller
-				if p.state.GetCurrentRound() < msg.View.Round {
-					// update timer
-					sendRoundChange(msg.View.Round)
-				}
-			}
-		} else {
-			totalVotingPower := TotalVotingPower(p.config.VotingPower)
-			validVotingPower := QuorumSizeVP(totalVotingPower)
-			roundVotingPower := p.state.calculateMessagesVotingPower(p.state.roundMessages[msg.View.Round], p.config.VotingPower)
-			if roundVotingPower >= validVotingPower {
-				// start a new round inmediatly
-				p.state.SetCurrentRound(msg.View.Round)
-				p.setState(AcceptState)
-			} else if roundVotingPower >= MaxFaultyVP(totalVotingPower)+1 {
-				// weak certificate, try to catch up if our round number is smaller
-				if p.state.GetCurrentRound() < msg.View.Round {
-					// update timer
-					sendRoundChange(msg.View.Round)
-				}
+		_ = p.state.AddRoundMessage(msg)
+
+		currentVotingPower := p.votingMetadata.CalculateVotingPower(p.state.roundMessages[msg.View.Round].extractNodeIds())
+		// Round change quorum is 2*F round change messages (F denotes max faulty voting power)
+		requiredVotingPower := 2 * p.votingMetadata.MaxFaultyVotingPower()
```

Comment:
We need p.QuorumSize there, don't we(if i'm not mistaken)?

Reply:
It is as it was before: for some reason quorum for round change message is `2*F` and not `QuorumSize`. Therefore didn't want to change that on my own.